### PR TITLE
Qiita VM OIDC login and admin page

### PR DESCRIPTION
**Pull request by Anna-Rehm**
The current status of the VM Qiita instance to use OIDC login and an admin page for user management following registration via OIDC

### CODE REVIEW

**Comment by sjanssen2** [comment]
I just pulled your code on top of the dev version and start to fire up the server, however it complains about the missing config section:
``` 
    raise NoSectionError(section) from None
configparser.NoSectionError: No section: 'oidc'
```

This is probably due to me not editing the config, but I think we should NOT make this section mandatory.

File: qiita_pet/handlers/auth_handlers.py [hunk 1]
```diff
@@ -165,11 +185,220 @@ def post(self):
...
         if user:
-            self.set_secure_cookie("user", json_encode(user))
+            self.set_secure_cookie("user", json_encode(user), SameSite=None)
         else:
             self.clear_cookie("user")
 
 
+
+class KeycloakMixin(OAuth2Mixin):
+    _OIDC_CLIENT_ID = '%s' % qiita_config.oidc_clients['KEYCLOAK_1']['OIDC_CLIENT_ID']
```
Comment (sjanssen2) [inline qiita_pet/handlers/auth_handlers.py:195]: KEYCLOAK_1 is not necessarily in the config file. Kills my qiita at the moment

**Comment by Anna-Rehm** [comment]
> I just pulled your code on top of the dev version and start to fire up the server, however it complains about the missing config section:
> 
> ```
>     raise NoSectionError(section) from None
> configparser.NoSectionError: No section: 'oidc'
> ```
> 
> This is probably due to me not editing the config, but I think we should NOT make this section mandatory.

That's true! I did not take that into consideration.

File: qiita_pet/handlers/auth_handlers.py (hunk 1 above)
Comment (Anna-Rehm) [reply qiita_pet/handlers/auth_handlers.py:195]: Thank you for pointing that out. I forgot to add this change, but any placeholder value should work for starting the server.

**Comment by sjanssen2** [comment]
closing in favor of #6 However, this PR was really instructive for me. Thanks!

**Pull request closed without merging**